plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -59,6 +60,15 @@ import Data.Text qualified as T
 import Data.Text.Encoding qualified as TE
 import Data.Traversable
 
+-- Patched GHC from haskell.nix
+-- with experimental core interface files patch,
+-- and a fix for unboxed tuples in GHCi for HLS support.
```

Comment:
```suggestion
{-
The patched 8.10 GHCs from haskell.nix include a buggy patch that introduced some changes to the core
AST that should only have been in 9.0. In order to facilitate building with non-patched GHCs, we use a bit 
of CPP here.

How do we know whether we're using a haskell.nix GHC? Well, we can't in general, but in fact we're using
an extra-special patched version which we *can* detect, so that lets us do the job for now until we can
fix this properly.
-}
```

Reply:
Thanks, I've applied it. I guess this works until the next GHC update now?

Reply:
Yeah, it's fragile. I was actually thinking of moving us to one of the unpatched GHCs at which point this would stop working, but it's okay for now.

Reply:
Yeah, I think it's easy enough to catch and fix this one when the time comes. I'll double-check every full moon too.